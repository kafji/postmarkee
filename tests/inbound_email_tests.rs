use postmarkee::InboundEmail;

#[test]
fn test_deserialize() {
    let serialized = include_str!("./postmark_inbound_payload_example.json");

    let email: InboundEmail = serde_json::from_str(serialized).unwrap();

    insta::assert_debug_snapshot!(email);
}

#[test]
fn test_serialize() {
    let email: InboundEmail = {
        let serialized = include_str!("./postmark_inbound_payload_example.json");
        serde_json::from_str(serialized).unwrap()
    };

    let serialized = serde_json::to_string_pretty(&email).unwrap();

    insta::assert_display_snapshot!(serialized);
}
