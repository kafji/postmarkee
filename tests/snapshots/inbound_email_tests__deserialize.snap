---
source: tests/inbound_email_tests.rs
assertion_line: 12
expression: email

---
InboundEmail {
    from_name: "Postmarkapp Support",
    message_stream: "inbound",
    from_full: Participant {
        email: "support@postmarkapp.com",
        name: "Postmarkapp Support",
        mailbox_hash: "",
    },
    to_full: [
        Participant {
            email: "yourhash+SampleHash@inbound.postmarkapp.com",
            name: "Firstname Lastname",
            mailbox_hash: "SampleHash",
        },
    ],
    cc_full: [
        Participant {
            email: "firstcc@postmarkapp.com",
            name: "First Cc",
            mailbox_hash: "",
        },
        Participant {
            email: "secondCc@postmarkapp.com",
            name: "",
            mailbox_hash: "",
        },
    ],
    bcc_full: [
        Participant {
            email: "firstbcc@postmarkapp.com",
            name: "First Bcc",
            mailbox_hash: "",
        },
        Participant {
            email: "secondbcc@postmarkapp.com",
            name: "",
            mailbox_hash: "",
        },
    ],
    original_recipient: "yourhash+SampleHash@inbound.postmarkapp.com",
    subject: "Test subject",
    message_id: "73e6d360-66eb-11e1-8e72-a8904824019b",
    reply_to: "replyto@postmarkapp.com",
    mailbox_hash: "SampleHash",
    date: 2014-08-01T20:45:32Z,
    text_body: "This is a test text body.",
    html_body: "<html><body><p>This is a test html body.</p></body></html>",
    stripped_text_reply: "This is the reply text",
    tag: "TestTag",
    headers: [
        Header {
            name: "X-Header-Test",
            value: "",
        },
        Header {
            name: "X-Spam-Status",
            value: "No",
        },
        Header {
            name: "X-Spam-Score",
            value: "-0.1",
        },
        Header {
            name: "X-Spam-Tests",
            value: "DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,SPF_PASS",
        },
    ],
    attachments: [
        Attachment {
            name: "test.txt",
            content: b"This is attachment contents, base-64 encoded.",
            content_type: "text/plain",
            content_length: 45,
        },
    ],
}
